sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/AdtQueryBuilder.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text;
+using Azure.DigitalTwins.Core.QueryBuilder;
+
+namespace Azure.DigitalTwins.Core.Queries.QueryBuilders
+{
+    /// <summary>
+    /// Azure DigitalTwins Query builder that facilitates writing queries against ADT instances.
+    /// </summary>
+    public class AdtQueryBuilder
+    {
+        private readonly SelectQuery _selectQuery;
+        private readonly FromQuery _fromQuery;
+        private readonly WhereQuery _whereQuery;
+
+        /// <summary>
+        /// Initializes an instance of an AdtQueryBuilder object.
+        /// </summary>
+        public AdtQueryBuilder()
+        {
+            _whereQuery = new WhereQuery(this);
+            _fromQuery = new FromQuery(this, _whereQuery);
+            _selectQuery = new SelectQuery(this, _fromQuery);
+        }
+
+        /// <summary>
+        /// Specifies the list of columns that the query will return
+        /// </summary>
+        /// <param name="args"> The arguments that can be queried (eg *, somePropertyName, etc.) </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery Select(params string[] args)
+        {
+            return _selectQuery.Select(args);
+        }
+
+        /// <summary>
+        /// Used to return only a certain number of top items that meet the query requirements.
+        /// </summary>
+        /// <param name="count"> The argument for TOP(), i.e the number of instances to return. </param>
```

Comment:
Referring to `TOP()` might look like part of our API surface, so be sure to specify this is the equivalent of `TOP()` in the SQL-like language that we're building.